cli/src/test/java/keywhiz/cli/commands/RenameActionTest.java

```diff
@@ -0,0 +1,75 @@
+package keywhiz.cli.commands;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import keywhiz.api.model.SanitizedSecret;
+import keywhiz.cli.configs.RenameActionConfig;
+import keywhiz.client.KeywhizClient;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class RenameActionTest {
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock KeywhizClient keywhiz;
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
I keep going back and forth on this - in the past, I have made some PRs where I replaced `expected=...` with `assertThatThrownBy` - e.g. https://github.com/squareup/java/pull/209020

I think the main advantage is that you can verify the contents of the exception message, but unless you have a huge reason do that, this is probably ok.

Reply:
You reminded me that the ExpectedException rule exists, I'll switch to that. In JUnit 5 this is much easier since it has lambda support so you can just assertThrows(). The downside of doing this at the annotation level is that if _any_ code in the test (not just the specific invocation you're targeting) throws the expected exception it'll pass . It is a rare case of tests passing trivially that I've run into at least once in my career.